MonkeyWrench.Web.WebService/WebServices.asmx.cs

```diff
@@ -161,18 +176,30 @@ public void AttachFileToLane (WebServiceLogin login, int lane_id, int lanefile_i
 		public void DeattachFileFromLane (WebServiceLogin login, int lane_id, int lanefile_id)
 		{
 			using (DB db = new DB ()) {
-				VerifyUserInRole (db, login, Roles.Administrator);
-				db.Audit (login, "WebServices.DeattachFileFromLane (lane_id: {0}, lanefile_id: {1})", lane_id, lanefile_id);
+				var lane = DBLane_Extensions.Create (db, lane_id);
+				VerifyUserInRoles (db, login, lane.additional_roles, false);
+				db.Audit (login, "WebServices.DeattachFileFromLane (lane_id: {0}, lanefile_id: {1})", lane.id, lanefile_id);
 				using (IDbCommand cmd = db.CreateCommand ()) {
 					cmd.CommandText = "DELETE FROM Lanefiles WHERE lane_id = @lane_id AND lanefile_id = @lanefile_id;";
-					DB.CreateParameter (cmd, "lane_id", lane_id);
+					DB.CreateParameter (cmd, "lane_id", lane.id);
 					DB.CreateParameter (cmd, "lanefile_id", lanefile_id);
 					cmd.ExecuteNonQuery ();
 				}
 			}
 		}
 
 		[WebMethod]
+		public void EditCommand (WebServiceLogin login, DBCommand command, int lane_id)
```

Comment:
Web services doesn't support overloading methods, so please use a different name (and the same goes for all the other new methods).